### PR TITLE
ensure that requests are retried and timeout on startup

### PR DESCRIPTION
**Pull request by charliesignalfx**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -629,8 +629,15 @@ func (p *gateway) addEndpoints(f protocol.DebugEndpointer, endpoints map[string]
 	}
 }
 
-func setClusterName(etcdClient etcdIntf.Client, clusterName string) (err error) {
-	_, err = etcdClient.Put(context.Background(), "/gateway/cluster/name", clusterName)
+func (p *gateway) setClusterName(ctx context.Context, etcdClient etcdIntf.Client, clusterName string) (err error) {
+	for ctx.Err() == nil {
```
Comment (mpetazzoni) [inline main.go]: Is this an infinite retry?

Comment (charliesignalfx) [reply main.go]: @mpetazzoni No, the context has a timeout on it.  When the timeout expires ctx.Err() will return an error that says "context deadline exceeded"

Comment (charliesignalfx) [reply main.go]: I can make the timeout duration a param and pass it in and then create a child context with timeout

Comment (mpetazzoni) [reply main.go]: No that's fine, thanks

**Comment by mpetazzoni** [comment]
@charliesignalfx are we still doing anything with this change?

**Comment by charliesignalfx** [comment]
Just rebased this.

**Pull request merged**